lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -437,12 +439,21 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             PerformanceEvents.InitializeBaseRequest,
             this.correlationId
         );
+
+        let configResponseMode = null;
+        if(this.config.auth.OIDCOptions.serverResponseType === ServerResponseType.FRAGMENT) {
+            configResponseMode = ResponseMode.FRAGMENT;
+        }
+        else {
+            configResponseMode = ResponseMode.QUERY;
+        }
```

Comment:
I'm running into a small problem with this one. Since the user can choose to set only one part of OIDCOptions (e.g. just the default scopes), OIDCOptions.serverResponseType is allowed to be either undefined or of type ServerResponseType. undefined does not map to ResponseMode. So even though, at this point, OIDCOptions.serverResponseType is defined because of the changes you suggested to Configuration.ts, Typescript throws an error because it has the potential to be undefined.

Reply:
Ah, this is because the `BrowserConfiguration` type makes every member of `auth` required but it's not recursive. You can either mess around with the types in the Configuration file to make sure nested objects are also wrapped in a `Required` utility type or, an easier way to address this, might be to make `OIDCOptions` its own category (sibling of auth, cache, system, etc)

Reply:
You can define a type like this:

```js
type InternalAuthOptions = Required<BrowserAuthOptions> & {
    OIDCOptions: Required<OIDCOptions>;
};
```

And replace existing references to `Required<BrowserAuthOptions>` with the new `InternalAuthOptions`.

Let me know if that doesn't work or if you want to work through this together